pkg/generic/httpthrift_codec.go

```diff
@@ -46,15 +46,37 @@ type HTTPRequest = descriptor.HTTPRequest
 type HTTPResponse = descriptor.HTTPResponse
 
 type httpThriftCodec struct {
-	svcDsc           atomic.Value // *idl
-	provider         DescriptorProvider
-	codec            remote.PayloadCodec
-	binaryWithBase64 bool
+	svcDsc              atomic.Value // *idl
+	provider            DescriptorProvider
+	codec               remote.PayloadCodec
+	binaryWithBase64    bool
+	enableDynamicgoReq  bool
+	enableDynamicgoResp bool
+	convOpts            conv.Options
 }
 
-func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec) (*httpThriftCodec, error) {
+func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec, opts ...Options) (*httpThriftCodec, error) {
 	svc := <-p.Provide()
-	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	var c *httpThriftCodec
+	if opts == nil {
+		c = &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	} else {
+		// codec with dynamicgo
+		convOpts := opts[0].DynamicgoConvOpts
+		if opts[0].EnableBasicDynamicgoConvOpts {
+			convOpts.EnableHttpMapping = true
+			convOpts.EnableValueMapping = true
+			convOpts.WriteRequireField = true
+			convOpts.WriteDefaultField = true
+			convOpts.OmitHttpMappingErrors = true
+			convOpts.NoBase64Binary = true
+		}
+		binaryWithBase64 := false
+		if !convOpts.NoBase64Binary {
+			binaryWithBase64 = true
+		}
+		c = &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: binaryWithBase64, enableDynamicgoReq: true, enableDynamicgoResp: opts[0].EnableDynamicgoHTTPResp, convOpts: convOpts}
```

Comment:
Why enableDynamicgoResp is configurable while enableDynamicgoReq is enabled by default？

Reply:
@jayantxie It's because the type of resp can be different between generic call using dynamicgo and one with fallback.
The resp type from generic call with dynamicgo will be json []byte (GeneralBody in HTTPResponse), while fallback one will be map[string]interface{} (Body in HTTPResponse).
So I added this enableDynamicgoResp flag in case the user wants to receive the response in a map like the original generic call.

Reply:
I see.